src/mongodb/scores.controller.ts

```diff
@@ -862,35 +3043,124 @@ export class ScoresController {
       }
     },
   })
-  GetFamiliarityScoreLearnerBySession(@Param('sessionId') id: string) {
-    return this.scoresService.getFamiliarityLearnerBySession(id);
+  GetConfidentVectorLearnerBySession(@Param('sessionId') id: string) {
+    return this.scoresService.getConfidentVectorBySession(id);
+  }
+
+  @ApiExcludeEndpoint(true)
+  @Post('/GetMissingChars')
+  async GetMissingChars(@Res() response: FastifyReply, @Body() storyData: any) {
+    let data = await this.scoresService.getMissingChars(storyData.storyLanguage);
+
+    let storyString = storyData.storyString;
+
+    let tokenArr = storyString.split("");
+
+    let taVowelSignArr = [
+      "ா",
+      "ி",
+      "ீ",
+      "ு",
+      "ூ",
+      "ெ",
+      "ே",
+      "ை",
+      "ொ",
+      "ோ",
+      "ௌ",
+      "்",
+    ];
+
+    let vowelSignArr = taVowelSignArr;
+
+    let uniqueChar = new Set();
+    let uniqueCharArr = [];
+    let prevEle = '';
+    let isPrevVowel = false;
+
+    // Create Unique token array
+    for (let tokenArrEle of tokenArr) {
+
+      for (let keyEle of tokenArrEle.split("")) {
+        if (vowelSignArr.includes(keyEle)) {
+          if (isPrevVowel) {
+            let prevEleArr = prevEle.split("");
+            if (prevEleArr.length) {
+              prevEle = prevEleArr[0] + keyEle;
+              uniqueCharArr[uniqueCharArr.length - 1] = prevEle;
+            }
+          } else {
+            prevEle = prevEle + keyEle;
+            uniqueCharArr[uniqueCharArr.length - 1] = prevEle;
+            //uniqueCharArr.push(prevEle);
+          }
+          isPrevVowel = true;
+        } else {
+          if (keyEle != ' ') {
+            uniqueCharArr.push(keyEle);
+          }
+          prevEle = keyEle;
+          isPrevVowel = false;
+        }
+      }
+    }
+
+    //let uniqueCharArr = Array.from(uniqueChar);
+    let matched = uniqueCharArr.filter(element => data.includes(element));
+    let matchtedTotal = matched.length;
+
+    let notIncluded = data.filter(element => {
+      if (!uniqueCharArr.includes(element)) {
+        return element;
+      }
+    });
+    let notIncludedTotal = notIncluded.length;
+
+    console.log(uniqueCharArr);
+    return response.status(HttpStatus.CREATED).send({ status: 'success', matched: matched, matchtedTotal: matchtedTotal, notIncluded: notIncluded, notIncludedTotal: notIncludedTotal })
+  }
+
+  @ApiExcludeEndpoint(true)
+  @Post('/addAssessmentInput')
+  async AddAssessmentInput(@Res() response: FastifyReply, @Body() assessmentInput: AssessmentInputDto) {
+    let data = await this.scoresService.assessmentInputCreate(assessmentInput);
+    return response.status(HttpStatus.CREATED).send({ status: 'success', msg: "Successfully stored data to Assessment Input" })
   }
+
+  @ApiExcludeEndpoint(true)
+  @Get('/GetSessionIds/:userId')
+  GetSessionIdsByUser(@Param('userId') id: string, @Query() { limit = 5 }) {
+    return this.scoresService.getAllSessions(id, limit);
+  }
+
```

Comment:
Overall, 
1. exception errors should be logged to Telemetry or Audit log.
2. Review use of each API and remove anything that is not relevant in the current scenario.
3. documentation is critical to understand the flow.
4. No Hardcoding in the code.
5. Keep it generic for future use and other language extenstions